lib/dialogs/trash-tag-confirmation/index.tsx

```diff
@@ -0,0 +1,60 @@
+import actions from 'lib/state/actions';
+import React, { FunctionComponent } from 'react';
+import { connect } from 'react-redux';
+
+import { closeDialog, tagToTrash } from '../../../lib/state/ui/actions';
+import Dialog from '../../dialog';
+
+import type * as S from '../../state';
+import type * as T from '../../types';
+
+type StateProps = {
+  tagToTrash: T.TagName;
+};
+
+type DispatchProps = {
+  closeDialog: () => any;
+  trashTag: (tagName: T.TagName) => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+const TrashTagConfirmation: FunctionComponent<Props> = ({
+  closeDialog,
+  tagToTrash,
+  trashTag,
+}) => (
+  <Dialog
+    className="trash-tag-confirmation"
+    onDone={closeDialog}
+    title="Delete Tag"
+  >
+    <div>Are you sure you want to delete this tag?</div>
+    <button
+      className="button-primary delete-tag"
+      onClick={() => {
+        trashTag(tagToTrash);
+        closeDialog();
```

Comment:
we want to avoid this kind of "double-dispatch" because it introduces _partial updates_ to our app, where the UI state is in transition. much better is to look at the dialog reducer and ask ourselves, "what changes this data?" we can answer that by saying, "if we trash a tag, we imply that we should close the tag-trash-confirmation dialog."

in that case we only need to dispatch the single action which expresses the user-oriented intent "yes, trash this tag," and let the application _react_ to that however it wants.

from time to time we may only want to conditionally do something like this, like close the dialog on trashing a tag. when those cases arise we might want to include metadata on the Redux action to indicate to the appropriate reducer that yes indeed this implies closing the dialog. here I'm not sure that extra metadata would be warranted because why would we continue showing that confirmation dialog if the tag was trashed.

that raises another point concerning the dialog reducer. _if_ it's open and the tag is trashed from another device we might want to close the dialog, or if we want to keep it open then we could have the dialog change to reflect that the tag was just remotely trashed